src/common/k8s-api/kube-object.store.ts

```diff
@@ -457,6 +457,9 @@ export abstract class KubeObjectStore<T extends KubeObject> extends ItemStore<T>
     const items = this.getItems();
 
     for (const { type, object } of this.eventsBuffer.clear()) {
+      // TODO: Convert items from Array to Map to avoid doing findIndex many times per second (events delay)
+      // Also, its worth to use Web Worker for such operations to prevent blocking UI
```

Comment:
Web worker for the watching would be great! Though updating the state probably would have to be done in the UI thread.

Reply:
Yes, I believe converting store items to maps will speed things up.